core/src/main/java/org/frankframework/processors/InputOutputPipeProcessor.java

```diff
@@ -166,8 +166,16 @@ private void processMessageCompaction(IPipe pipe, PipeLineSession pipeLineSessio
 			return;
 		}
 		InputSource inputSource = getInputSourceFromResult(result, pipe, owner);
-		XmlWriter xmlWriter = new XmlWriter();
-		CompactSaxHandler handler = new CompactSaxHandler(xmlWriter);
+
+		final MessageBuilder messageBuilder;
+		try {
+			messageBuilder = new MessageBuilder();
+		} catch (IOException e) {
+			log.warn("Pipeline of adapter [{}] could not compact received message", owner.getName(), e);
+			return;
+		}
+
```

Comment:
Why not put everything in the try-catch clause below?
(I don't like the way it looks with this separate try-catch here).